src/backend/distributed/commands/utility_hook.c

```diff
@@ -1627,17 +1664,93 @@ RunPreprocessMainDBCommand(Node *parsetree, const char *queryString)
 static void
 RunPostprocessMainDBCommand(Node *parsetree)
 {
+	if (IsStatementSupportedInNonMainDb(parsetree) &&
+		StatementRequiresMarkDistributedFromNonMainDb(parsetree))
+	{
+		MarkObjectDistributedInNonMainDb(parsetree);
+	}
+}
+
+
+/*
+ * GetObjectInfo returns the name and oid of the object in the given parsetree.
```

Comment:
Given that what ObjectInfo stores might change over time, let's generalize the comment:

```suggestion
 * GetObjectInfo returns ObjectInfo for the target object of given parsetree.
```